plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -619,6 +621,49 @@ public void testExternalTableFilesRetainedOnDrop()
         assertEquals(getTableFiles(tableName).size(), fileCount);
     }
 
+    @Test
+    public void testRenameExternalTable()
+    {
+        String oldTable = "test_external_table_rename_old_" + randomTableSuffix();
+        String tableLocation = getLocationForTable(bucketName, oldTable);
+        assertUpdate(
+                format("CREATE TABLE %s (a bigint, b double) WITH (location = '%s')", oldTable, tableLocation));
+
+        String newTable = "test_rename_new_" + randomTableSuffix();
+        assertUpdate("ALTER TABLE " + oldTable + " RENAME TO " + newTable);
+
+        assertThat(query("SHOW TABLES LIKE '" + oldTable + "'"))
+                .returnsEmptyResult();
+        assertThat(query("SELECT a, b FROM " + newTable))
+                .returnsEmptyResult();
+
+        assertUpdate("INSERT INTO " + newTable + " (a, b) VALUES (42, -38.5)", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + newTable))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+
+        assertUpdate("DROP TABLE " + newTable);
+        assertThat(query("SHOW TABLES LIKE '" + newTable + "'"))
+                .returnsEmptyResult();
+    }
+
+    @Test
+    public void testRenameManagedTableIsNotSupportedWhenRenameManagedTablesNotEnabled()
```

Comment:
Do you have a test that shows the underlying files are actually moved when renaming managed tables?

Reply:
The default `public void testRenameTable()` from `BaseConnectorSmokeTest` tests renaming of a managed table. Now we don't actually verify if the files are moved, do you think that's useful? In the end you'll be testing what hive metastore is doing. If hive metastore starts doing something different, the read query would fail. We depend on the SERDE location in the storage descriptor for the reads.

Reply:
Right, I forgot about the base class tests. That's fine with me